### ISSUE TITLE
Debug output messes up AgdaInfo buffer

### ISSUE THREAD

**Issue by nad** [agda/agda#2602]
Consider the following code:
```agda
{-# OPTIONS -vprofile:7 #-}

A : Set
A = {!Set!}
```
When this code is loaded in Emacs and the expression in the hole is "given" I get the following output:
```
[...]/P.agda:4,7-10
Set₁ != Set
when checking that the expression Set has type SetTotal 0ms 

```
Note that the debug output is placed right after the error message, without intervening whitespace. There is also a second problem: The cursor is positioned at the end of the output, rather than at the beginning (as is usually the case), meaning that (at least with my current setup) I don't see the start of the error message unless I take some action.

This is a regression, Agda 2.5.2 does not even display the debug output in the `AgdaInfo` buffer. @fredrikNordvallForsberg, does this have something to do with your work on warnings?

Previously debug output was placed in a separate buffer, with new content appended to the end of the buffer, and the cursor (intentionally) placed at the end. I would prefer to restore the old behaviour, I don't want to see debug output in the `AgdaInfo` buffer.

**Comment by fredrikNordvallForsberg** [comment]
I think this is related to the following change by @UlfNorell : https://github.com/agda/agda/commit/a46fbe4efc072d41bfd995f7adff4b8810cfe8a3#diff-6ecdb7c4d91fa74f5e59ab3c278e38e6R70

If I revert that line to `reportSLn benchmarkKey benchmarkLevel $ prettyShow b` I get the old behaviour using the `Agda Debug` buffer.

**Comment by nad** [comment]
@UlfNorell, was this an intended change?

**Comment by UlfNorell** [comment]
It's certainly desirable to get the time measurements from `-v profile:interactive:10` in the info buffer. Typically I want to run `C-c C-n` on some code under optimisation and see how long it takes. It would break my workflow very much if I had to switch back and forth between the code and the debug buffer every time. The other profiling data should most likely go in the debug buffer.

**Comment by nad** [comment]
From `Agda.Interaction.EmacsTop`:
```
lispifyResponse (Resp_RunningInfo n s)
  | n <= 1    = return [ displayRunningInfo s ]
  | otherwise = return [ L [A "agda2-verbose", A (quote s)] ]
```
@UlfNorell's hack set the verbosity level to 0, thus directing the output to the info buffer. The benchmarking information isn't erased, because it is output after the goals or type error have been displayed. However, the formatting and focus is messed up. Now I also noticed that the mode line is messed up, it displays `*Type-checking*` rather than (say) `*All Done*`.

I suggest that we either revert @UlfNorell's change, or do something more principled. @fredrikNordvallForsberg, could your warning machinery be used for benchmarks as well? Would that be appropriate?

**Comment by fredrikNordvallForsberg** [comment]
> @fredrikNordvallForsberg, could your warning machinery be used for benchmarks as well? Would that be appropriate?

The naive solution of just changing the `reportSLn` into raising a warning instead does not work since the printing of the benchmarking information happens after the printing of the warnings. It shouldn't be hard to do something principled though, right?

**Comment by gallais** [comment]
We could add a new category: `InfoWarning` to really distinguish these from things that are actually problematic.

**Comment by andreasabel** [comment]
> I think this is related to the following change by @UlfNorell : a46fbe4#diff-6ecdb7c4d91fa74f5e59ab3c278e38e6R70
> If I revert that line to `reportSLn benchmarkKey benchmarkLevel $ prettyShow b` I get the old behaviour using the Agda Debug buffer.

I implemented this fix for the release of 2.5.3 (see #2659).
